Duplicati/Library/Main/Controller.cs

```diff
@@ -111,7 +111,13 @@ public Duplicati.Library.Interface.IBackupResults Backup(string[] inputsources,
             Library.UsageReporter.Reporter.Report("USE_BACKEND", new Library.Utility.Uri(m_backend).Scheme);
             Library.UsageReporter.Reporter.Report("USE_COMPRESSION", m_options.CompressionModule);
             Library.UsageReporter.Reporter.Report("USE_ENCRYPTION", m_options.EncryptionModule);
-            
+
+            if (!m_options.NoAutoCompact && (LastCompact > DateTime.MinValue) && (LastCompact.Add(m_options.AutoCompactInterval) > DateTime.Now))
+            {
+                Logging.Log.WriteInformationMessage(LOGTAG, "CompactResults", "Skipping auto compaction until {0}", LastCompact.Add(m_options.AutoCompactInterval));
+                m_options.RawOptions["no-auto-compact"] = "true";

```

Comment:
It feels a bit strange to be toggling the `no-auto-compact` option to achieve the desired behavior.  Would it make more sense to modify the [BackupHandler.CompactIfRequired](https://github.com/duplicati/duplicati/blob/master/Duplicati/Library/Main/Operation/BackupHandler.cs#L243) method?  That way, the logic that determines when the delete or compact operations occur reside in the same place.

Reply:
At first I was going to put it there, but then I saw that the DeleteHandler DoRun may also call Compact.  Intent of this change was not to defer the Deletion phase - just the Compaction phase.

So I was hunting for a single location where I could put the check.

I think there was a second reason that CompactIfRequired wasn't ideal but I cannot recall at the moment.

Reply:
Is the `CompactHandler.DoCompact` method a feasible location?

Reply:
Problem with that spot is we get there for intentional compact OR auto compact.  Need to be able to differentiate how we got there.  Also, we would have to make that determination even before db.GetCompactReport (first line of function) because that is what ultimately calls the GetWastedSpaceReport which is the root reason for this change - that WastedSpace report is what takes a long time on some systems.

That's ultimately why I thought temporarily flipping the no-auto-compact option near the start of the entire backup job was a decent approach.

What do you think?

Reply:
Hmmm...what are the different scenarios where compact can occur, and which ones are you trying to skip with this option?  I can think of 
* Manually run (via command-line or UI)
* Backup operation
* Delete operation
* Purge operation

Reply:
Those are the only three that I know of, and the intent of this change was to defer ANY automatic compaction so they don't happen too frequently.  Intentional compact operation will still run.

The alternative to flipping the m_options value temporarily would be to add an additional check to see if it should be deferred in all three of these locations. I could try to rework it that way if you think it would be better.

Reply:
Not sure if this will affect your opinion, but thought I'd mention in case you weren't aware.... the use of RawOptions to temporarily change a flag is not unprecedented. In fact I got the idea from a usage elsewhere in the code.  See `PurgeBrokenFilesHandler.cs`  (in fact they change the same no-auto-compact flag there)

So maybe it's not so bad to use this method? ;)

Reply:
Yes, I just saw that as well.  I think what you have here may be the cleanest implementation. :+1: 

One thing I would eventually like to pursue is to rewrite how we handle all the options.  For example, it would be nice if we can hide the internal `Dictionary` and avoid knowledge of the underlying strings elsewhere in the code.